scripts/pylib/twister/twisterlib/runner.py

```diff
@@ -511,6 +517,47 @@ def process(self, pipeline, done, message, lock, results):
             else:
                 self.cleanup_artifacts()
 
+    def amend_test_instance(self, results):
```

Comment:
why amend? The goal is to fuilly rely on what comes out of the symbols file as the list of testcases in the suite. This should just replaces whatever is in there and should be seen as the true source.

At this stage we should also not skip anything and should make judgment about unintended tests, this should purly serve as a replacement for regex method, however, only for ztest with new API for now.

We also want to capture the testsuite names from the symbols file.

Reply:
Please see my next reply.